src/arrayMethodJoin.js

```diff
@@ -5,7 +5,46 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    switch (this[0]) {
+      case null:
+        break;
+
+      case undefined:
+        return result;
+
+      default:
+        result += this[0];
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      switch (separator) {
+        case '':
+          result += this[i];
+          break;
+
+        case null:
+          result += 'null';
+          result += this[i];
+          break;
+
+        case (undefined):
+          result += ',';
+          result += this[i];
+          break;
+
+        default:
+          if (this[i] === null || this[i] === undefined) {
+            result += separator;
+            break;
+          }
+          result += separator;
+          result += this[i];
+      }
+    }
```

Comment:
try to simplify your solution.
switch cases are redundant.
enough to make 2 checks inside loop
1. this[i] not equal null and this[i] not equal undefined
2. it is not last element.